omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/summary/RegistrationSummaryFragmentController.java

```diff
@@ -62,6 +80,23 @@ public void controller(FragmentConfiguration config,
         model.addAttribute("overallActions", overallActions);
 
     }
-
+    
+    protected List<Extension> filter(List<Extension> extensionList, final String extensionPointId) {
+    	
+    	List<Extension> evaluatedExtensionList = new ArrayList<Extension>();
+    	evaluatedExtensionList.addAll(extensionList);
+    	
+    	CollectionUtils.filter(evaluatedExtensionList, new Predicate<Extension>() {
+            
+       	 @Override
+         public boolean evaluate(Extension extension) {
+       	    if(extensionPointId.equals(extension.getExtensionPointId())) {
+	              return true;
+	            }
+	            return false;
+	         }
+	    });
+    	return evaluatedExtensionList; 
+    }
```

Comment:
Let's give it some JavaDoc and remove the redundant variable:
```java
/**
 * Filters an extension list by selecting extensions that match a given extension point ID.
 */
protected List<Extension> filter(List<Extension> extensionList, final String extensionPointId) {

  List<Extension> ret = new ArrayList<Extension>();
  ret.addAll(extensionList);

  CollectionUtils.filter(ret, new Predicate<Extension>() {
    @Override
    public boolean evaluate(Extension extension) {
      if(extensionPointId.equals(extension.getExtensionPointId())) {
        return true;
      }
      return false;
    }
  });
  return ret; 
}
```

Reply:
Resolved.